### PR TITLE
Feature/ptcloud proc

### PR DESCRIPTION
**Pull request by JmcRobbie**

### CODE REVIEW

File: nova_rover_demos/pointCloudProcessing/ptcloudgen.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		'''
+		xSteps = int(abs(self.size[0][0] -self.size[0][1])/x_res)
+		ySteps = int(abs(self.size[1][0] -self.size[1][1])/y_res)
+		obstacleCloud = self.ptcloud[np.where(self.ptcloud[:,2]>0.2)]
+		self.occupancy = np.zeros([xSteps,ySteps]) ## construct occupancy grid
+		for point in obstacleCloud:
+			xc = int(point[0]/x_res)
+			yc = int(point[1]/y_res)
+			self.occupancy[xc][yc] = self.occupancy[xc][yc] + point[2]
+			if self.occupancy[xc][yc]>10.0:
```
Comment (naverill) [inline nova_rover_demos/pointCloudProcessing/ptcloudgen.py]: Hmm I would set the max and min bounds of the point cloud to be optional parameters in the function to make it easier to configure later

File: nova_rover_demos/pointCloudProcessing/ptcloudgen.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Author: Jack McRobbie
+'''
+import math
+import numpy as np
+import random as rn
+from matplotlib import pyplot
+from mpl_toolkits.mplot3d import Axes3D
+
+class point_cloud_gen:
```
Comment (naverill) [inline nova_rover_demos/pointCloudProcessing/ptcloudgen.py]: Bruh what is this PEP8

**Review by naverill — APPROVED** [review]
Code looks good and I'm not getting any errors when running it but boi does this need a good refactor. Otherwise all good to go.

File: nova_rover_demos/pointCloudProcessing/ptcloudgen.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+	def __init__ (self,size, features,noise = 1,N = 10000):
+		'''
+		size: the bounds on the range of the point cloud values. Is a list of n tuples where n is the number of dimensions
```
Comment (naverill) [inline nova_rover_demos/pointCloudProcessing/ptcloudgen.py]: Also probably best to indicate the scale (m/cm) of the upper and lower bound in size

**Review by naverill — APPROVED** [review]

File: nova_rover_demos/pointCloudProcessing/ptcloudgen.py (hunk 3 above)
Comment (JmcRobbie) [reply nova_rover_demos/pointCloudProcessing/ptcloudgen.py]: good point, ill make a change.

File: nova_rover_demos/pointCloudProcessing/ptcloudgen.py (hunk 2 above)
Comment (JmcRobbie) [reply nova_rover_demos/pointCloudProcessing/ptcloudgen.py]: yeah - what is Pep8?

File: nova_rover_demos/pointCloudProcessing/ptcloudgen.py (hunk 1 above)
Comment (JmcRobbie) [reply nova_rover_demos/pointCloudProcessing/ptcloudgen.py]: great idea, ill make that change as well.

**Pull request merged**